src/lib.rs

```diff
@@ -17,65 +17,84 @@
 ///
 /// ```
 /// const fn signature() -> const_sha1::Digest {
-///     const_sha1::sha1(&const_sha1::ConstBuffer::from_slice(stringify!(MyType).as_bytes()))
+///     const_sha1::sha1(stringify!(MyType).as_bytes())
 /// }
 /// ```
-pub const fn sha1(data: &ConstBuffer) -> Digest {
+pub const fn sha1(data: &[u8]) -> Digest {
     let state: [u32; 5] = [0x67452301, 0xefcdab89, 0x98badcfe, 0x10325476, 0xc3d2e1f0];
-    let len: u64 = 0;
     let blocks = Blocks {
         len: 0,
         data: [0; 64],
     };
-    let (blocks, len, state) = process_blocks(blocks, data, len, state);
+    let (blocks, len, state) = process_blocks(blocks, data, data.len(), state);
     digest(state, len, blocks)
 }
 
-/// The size of the ConstBuffer
+/// A const evaluated sha1 function.
+///
+/// # Use
+///
+/// ```
+/// const fn signature() -> const_sha1::Digest {
+///     const_sha1::sha1_of_buffer(&const_sha1::ConstBuffer::from_slice(stringify!(MyType).as_bytes()))
+/// }
+/// ```
+pub const fn sha1_of_buffer(data: &ConstBuffer) -> Digest {
```

Comment:
Not sure I like the name of the function. Perhaps `sha1_from_const_buffer` is better? Maybe we should rename `ConstBuffer` to `ConstSlice` and name this function `sha1_from_const_slice`. Thoughts?

Reply:
That was a hardest part of the work, so I decided to leave it up to you, but forgot to mention it. Changing name to `sha1_from_const_slice` and renaming the struct sounds as the best solution.